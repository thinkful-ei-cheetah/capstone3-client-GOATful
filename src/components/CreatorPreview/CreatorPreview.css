.creator-preview {
  width: 100%;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.preview-controls {
  margin-bottom: 20px;
}

.preview-controls  > * {
  margin-right: 15px;
}

.device-selector {
  cursor: pointer;
  background-color: white;
  border: thin solid #C4302B;
  border-radius: 4px;
  display: inline-block;
  font: inherit;
  line-height: 1.5em;
  padding: 0.5em 3.5em 0.5em 1em;

  /* reset */

  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  background-image:
    linear-gradient(45deg, transparent 50%, gray 50%),
    linear-gradient(135deg, gray 50%, transparent 50%),
    radial-gradient(#ddd 70%, transparent 72%);
  background-position:
    calc(100% - 20px) calc(1em + 2px),
    calc(100% - 15px) calc(1em + 2px),
    calc(100% - .5em) .5em;
  background-size:
    5px 5px,
    5px 5px,
    1.5em 1.5em;
  background-repeat: no-repeat;
}

.device-selector:focus {
  background-image:
    linear-gradient(45deg, white 50%, transparent 50%),
    linear-gradient(135deg, transparent 50%, white 50%),
    radial-gradient(gray 70%, transparent 72%);
  background-position:
    calc(100% - 15px) 1em,
    calc(100% - 20px) 1em,
    calc(100% - .5em) .5em;
  background-size:
    5px 5px,
    5px 5px,
    1.5em 1.5em;
  background-repeat: no-repeat;
  outline: 0;
}

.randomize {
  padding: 11px 13px;
}
