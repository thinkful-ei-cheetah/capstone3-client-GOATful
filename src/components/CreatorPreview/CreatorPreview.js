import React, {Component} from 'react'
import './CreatorPreview.css'
import YoutubeSearchResult from '../YoutubeSearchResult/YoutubeSearchResult'
import YoutubeApiService from '../../services/youtube-api'
  
export default class CreatorPreview extends Component {
  state = {
    userPreview: {},
    youtubeSearchResults: [],
    userVideo: {}
  }

  async componentDidMount() {
    const video = JSON.parse(window.localStorage.getItem('public_user_video'))
    const results = await YoutubeApiService.search(video.tags)
    this.setState({
      userPreview: this.props.userPreview,
      youtubeSearchResults: [...results],
      userVideo: {...video}
    })
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.userPreview !== prevProps.userPreview) {
      this.setState({
        userPreview: this.props.userPreview
      })
    }
  }

  renderPreviews = () => {
    const {userPreview, youtubeSearchResults, userVideo} = this.state
    userPreview.youtube_display_name = userVideo.youtube_display_name
    userPreview.video_length = userVideo.video_length
    const videos = [userPreview, ...youtubeSearchResults]
    return videos.map((video, i) => {
      return <YoutubeSearchResult {...video} key={i}/>
    })
  }

  render() {
    return (
      <div className='creator-preview'>
        <h2>How Your Video Will Look</h2>
        <div className='preview-controls'>
        <select>
          <option value="desktop">Desktop</option>
          <option value="iphone">iPhone</option>
          <option value="android">Android</option>
        </select>
        <button className='button'>Randomize</button>
        </div>
        
        {this.renderPreviews()}
      </div>
    )
  }
}
  
// const videoData = {
//   title: 'My first test video',
//   tags: ['coding', 'interview', 'algorithm'],
//   youtube_display_name: 'Byte By Byte',
//   video_length: '8:56'
// }
// window.localStorage.setItem('public_user_video', JSON.stringify(videoData))