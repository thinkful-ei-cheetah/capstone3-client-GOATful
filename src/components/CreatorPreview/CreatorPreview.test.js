import React from 'react';
import ReactDOM from 'react-dom';
import CreatorPreview from './CreatorPreview';
import renderer from 'react-test-renderer';

describe('CreatorPreview Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< CreatorPreview />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<CreatorPreview/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
