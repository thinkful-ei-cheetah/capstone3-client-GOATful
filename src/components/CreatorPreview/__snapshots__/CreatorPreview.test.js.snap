// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorPreview Component renders the UI as expected 1`] = `
<div
  className="creator-preview"
  id="creator-preview"
>
  <h2>
    Video Project: 
  </h2>
  <div
    className="preview-controls"
  >
    <select
      className="device-selector"
      onChange={[Function]}
    >
      <option
        value="desktop"
      >
        Desktop
      </option>
      <option
        value="mobile"
      >
        Mobile
      </option>
    </select>
    <button
      className="randomize button"
      onClick={[Function]}
      title="change the ordering in which your thumbnail appears to ensure it stands out"
    >
      Randomize
    </button>
  </div>
  <div
    className="desktop-view-page"
    id="dvp"
  />
</div>
`;
